pandas/core/groupby/groupby.py

```diff
@@ -813,9 +813,10 @@ def _try_cast(self, result, obj, numeric_only: bool = False):
                 # datetime64tz is handled correctly in agg_series,
                 #  so is excluded here.
 
-                # return the same type (Series) as our caller
-                cls = dtype.construct_array_type()
-                result = try_cast_to_ea(cls, result, dtype=dtype)
+                if len(result) and isinstance(result[0], dtype.type):
```

Comment:
can u do this inside try_to_cast_ea?

i am ok with this generally

Reply:
In principle I agree. For now (1.0.0) I think I prefer keeping this change narrow.

I'm definitely going to open a followup issue with some more discussion / potential fixes.